lib/event-utils.js

```diff
@@ -0,0 +1,206 @@
+    }, object);
+}
+
+
+/**
+ * Given a URL, returns a Promise of the contents at that
+ * URL.  Supports both file:// (via fs.readFile) and other http
+ * based URLs with preq.get.
+ * @param {string} u
+ * @return {Promise<string>}
+ */
+function urlGet(u) {
+    if (_.isString(u)) {
+        u = new URL(u);
+    }
+
+    // Read from local filesystem if file:/// or no protocol is given.
+    if (u.protocol === 'file:') {
+        return readFile(u.pathname, 'utf-8');
+    } else {
+        return preq.get({ uri: u.href }).then(res => res.body);
+    }
+}
+
+/**
+ * Given a URL, returns a Promise of the contents at that
+ * converted into an Object.  The content at URL
+ * must either be a JSON or YAML string.
+ * @param {string} u
+ */
+function urlGetObject(u) {
+    return urlGet(u).then(content => objectFactory(content));
+}
+
+/**
+ * Returns the file extension (or the last part after a final '.' in a file basename)
+ * of a filename path.
+ * @param {string} filename
+ */
+function fileExtension(filename) {
+    if (!filename) {
+        return '';
+    }
+
+    const parts = basename(filename).split('.');
+    if (parts.length > 1) {
+        return parts[parts.length - 1];
+    } else {
+        return '';
+    }
+}
+
+const uriProtocolRegex = /^[a-z]+:\/\//;
+function uriHasProtocol(uri) {
+    return uriProtocolRegex.test(uri);
+}
+
+function resolveUri(uri, baseUri, defaultFileExtension) {
+    let url = uri;
+    // If uri doesn't already have an extension, and we are given a default one, append it.
+    if (!fileExtension(uri) && defaultFileExtension) {
+        url = uri + defaultFileExtension;
+    }
+
+    // If the uri doesn't have a protocol, then we can use
+    // the given baseUri as the default.
+    if (baseUri && !uriProtocolRegex.test(url)) {
+        url = baseUri + url;
+    }
+
+    // If the url still doesn't have a protocol, assume it should be file://.
+    if (!uriProtocolRegex.test(url)) {
+        url = `file://${path.resolve(url)}`;
+    }
+
+    // console.log(`Resolved ${uri} to ${url}`);
+    return url;
+}
+
+/**
+ * Given an event, extracts and returns a new URL
+ * @param {string} uriField field path in event to extract URI
+ * @param {string} baseUri If given, this will be prefixed to the extracted URI
+ * @param {string} defaultFileExtension If the basename of the URI does not end
+ *                 in a file extension, this will be appended to the URI.
+ * @param {Object} event The event to extract a URL from.
+ * @return {string} of the extracted URL
+ */
+function extractUrl(uriField, baseUri, defaultFileExtension, event) {
+    const uri = objectProperty(uriField, event);
+    return url.parse(resolveUri(uri)).href;
+}
+
+const defaultLoggerOptions = { src: true, level: 'info' };
+function loggerFactory(logger, options = {}) {
+    if (_.isUndefined(logger)) {
+        return bunyan.createLogger(Object.assign(defaultLoggerOptions, options));
+    } else {
+        return logger.child();
```

Comment:
What's the point in creating a child without providing any additiona options to it?

Reply:
Hm, maybe a good point.  I just wanted to DRY my logger creations in various places to be able to use the passed in one, or create one.  Maybe I don't need a child at all?

Reply:
You probably don't need a child at all.